pkg/frontend/v2/frontend.go

```diff
@@ -76,6 +77,7 @@ func (cfg *Config) RegisterFlags(f *flag.FlagSet, logger log.Logger) {
 	f.StringVar(&cfg.Addr, "query-frontend.instance-addr", "", "IP address to advertise to the querier (via scheduler) (default is auto-detected from network interfaces).")
 	f.IntVar(&cfg.Port, "query-frontend.instance-port", 0, "Port to advertise to querier (via scheduler) (defaults to server.grpc-listen-port).")
 
+	cfg.GRPCClientConfig.CustomCompressors = []string{s2.Name}
```

Comment:
uh, so we need to pass this to all clients? I somehow thought it would be enough to register it in dskit "globally".

Reply:
Maybe we can find a better solution after this PR?